app/experimenter/experiments/forms.py

```diff
@@ -434,19 +430,47 @@ def is_valid(self):
         return super().is_valid() and self.variants_formset.is_valid()
 
     def save(self, *args, **kwargs):
-        experiment = super().save(*args, **kwargs)
+        self.variants_formset.save()
+        return super().save(*args, **kwargs)
 
-        for form in self.variants_formset.forms:
-            if form.cleaned_data["DELETE"] and form.cleaned_data["id"]:
-                form.instance.delete()
-            elif form.is_valid():
-                form.instance.experiment = experiment
-                form.save(*args, **kwargs)
 
-        return experiment
+class ExperimentVariantsAddonForm(ExperimentVariantsBaseForm):
+
+    FORMSET_FORM_CLASS = ExperimentVariantAddonForm
+    FORMSET_CLASS = ExperimentVariantsFormSet
+
+    addon_name = forms.CharField(
+        required=False,
+        label="Add-On Name",
+        help_text=Experiment.ADDON_NAME_HELP_TEXT,
+    )
+    addon_experiment_id = forms.CharField(
+        required=False,
+        label="Add-On Experiment ID",
+        help_text=Experiment.ADDON_EXPERIMENT_ID_HELP_TEXT,
+    )
+    addon_testing_url = forms.URLField(
+        required=False,
+        label="Add-On Signed Testing Location",
```

Comment:
Out of curiosity, why the word "Location" and not URL? I tend to believe "Location" could add more confusion than "URL" but I'm also very web techy and perhaps blind to the nuances here.